test/thread-management.test.js

```diff
@@ -25,7 +25,7 @@ test('emit error if thread exits', async function (t) {
     sync: true
   })
 
-  stream.on('ready', function () {
+  queueMicrotask(() => {
```

Comment:
Can you move this and the following to the ready event?

Reply:
For some reason the tests fail then :(

Reply:
TBH I think ready prop and event just adds unnecessary complexity. Not sure why we need them.

Reply:
They were definitely needed before this PR. As of right now pino depends on this event being there.

Reply:
> For some reason the tests fail then :( 

Seems odd, likely a bug still :(.

Reply:
> As of right now pino depends on this event being there.

Got it. I'll try to maintain compat with previous behavior and maybe we can remove it in semver major. Maybe I'll prepare a pino pr at some later point.

Reply:
Fixed.